File: Vibe MessagesExtension/Models/Supporting/Streak.swift
//
//  Streak.swift
//  Vibe MessagesExtension
//
//  Created on 1/22/26.
//

import Foundation

struct Streak: Codable, Equatable {
    let conversationId: String
    let currentStreak: Int
    let longestStreak: Int
    let lastPostDate: Date?
    let todayPosters: [String]

    var hasPostedToday: Bool {
        guard let lastPost = lastPostDate else { return false }
        return Calendar.current.isDateInToday(lastPost)
    }

    func userPostedToday(_ userId: String) -> Bool {
        todayPosters.contains(userId)
    }
}

File: backend/src/models/Streak.ts
import mongoose, { Schema, Model } from 'mongoose';
import { IStreak } from '../types';

const streakSchema = new Schema<IStreak>(
  {
    conversationId: {
      type: String,
      required: true,
      unique: true,
    },
    currentStreak: {
      type: Number,
      default: 0,
    },
    longestStreak: {
      type: Number,
      default: 0,
    },
    lastPostDate: { type: Date },
    todayPosters: [{ type: String }],
  },
  { timestamps: true }
);

const Streak: Model<IStreak> = mongoose.model<IStreak>('Streak', streakSchema);

export default Streak;
